src/test/regress/sql/multi_extension.sql

```diff
@@ -130,6 +130,15 @@ ALTER EXTENSION citus UPDATE TO '9.2-4';
  */
 ALTER EXTENSION citus UPDATE TO '9.3-1';
 ALTER EXTENSION citus UPDATE TO '9.3-2';
+
+-- downgrade path from 9.3-2 to 9.2-2
+ALTER EXTENSION citus UPDATE TO '9.2-2';
+ALTER EXTENSION citus UPDATE TO '9.3-2';
```

Comment:
To test that the downgrade actually did something we should run:
```
\dx+ citus
```
The output of this should never change, otherwise the downgrade script does not work.

Reply:
Maybe we can just figure out the underlying query of \dx+ and compare the output before/after.

Reply:
Using the underlying query sounds fine to me. 
And I think for this PR it indeed makes sense to include both before and after, so it can be compared manually during review.

For the future though I think it makes more sense to add this query right and then let the regular `diff` make sure that this output is not changed by a PR.

Reply:
I introduced a new function that stores descriptions of all objects that depend on citus extension, and print out the diffs after an upgrade/downgrade operation.

I used the underlying query of `\dx+` as a starting point, and stored the list of descriptions on a table.

Reply:
I like the idea of looking at everything that depends on Citus, this makes sure that everything that got added in a new version is deleted with the downgrade!

Reply:
One of the changes in upgrade script are about changing the return type of a function that is not really covered in the current output because the return type is not part of the function signature in PG.

Similarly changes in the catalogue table records are not covered as well.

Reply:
I very much like this `print_extension_changes()` function, but I think it's more useful if it's called a bit differently.
```sql
ALTER EXTENSION citus UPDATE TO '9.2-2';
-- Snapshot of state at 9.2-2
SELECT * FROM print_extension_changes();

-- Test downgrade to 9.2-2 from 9.3-2
ALTER EXTENSION citus UPDATE TO '9.3-2';
ALTER EXTENSION citus UPDATE TO '9.2-2';
-- Should be empty result since upgrade+downgrade should be a no-op
SELECT * FROM print_extension_changes();

-- Upgrade to 9.3-2
ALTER EXTENSION citus UPDATE TO '9.3-2';
-- Snapshot of state at 9.3-2
SELECT * FROM print_extension_changes();

-- Test downgrade to 9.3-2 from 9.4-1
ALTER EXTENSION citus UPDATE TO '9.4-1';
ALTER EXTENSION citus UPDATE TO '9.3-2';
-- Should be empty result since upgrade+downgrade should be a no-op
SELECT * FROM print_extension_changes();
```
This way we won't have to manually compare the results from the query, because all we expect is an empty result.

Reply:
Jelte I really liked this approach. Now it's cleaner and easier to read.

While I was at it, I removed some extra tests for unreachable versions to keep things consistent.